src/runtime/virtcontainers/container.go

```diff
@@ -810,7 +839,14 @@ func (c *Container) createDevices(contConfig *ContainerConfig) error {
 	// If devices were not found in storage, create Device implementations
 	// from the configuration. This should happen at create.
 	var storedDevices []ContainerDevice
-	for _, info := range contConfig.DeviceInfos {
+	var deviceInfos []config.DeviceInfo
+	virtualVolumesDeviceInfos, err := c.createVirtualVolumeDevices()
+	if err != nil {
+		return err
+	}
+	deviceInfos = append(virtualVolumesDeviceInfos, contConfig.DeviceInfos...)
```

Comment:
`deviceInfos := append(virtualVolumesDeviceInfos, contConfig.DeviceInfos...)？

Reply:
`deviceInfos ` is defined in line 842

Reply:
Line 842 is redundant:)

Reply:
In the case of layer raw disks , there can be multiple devices; removing this line could lead to deviceinfo loss.

Reply:
Fixed.